src/shell/notification.js

```diff
@@ -229,84 +230,72 @@ const Source = GObject.registerClass({
 
         // Check if this is a repeat
-        if (notification) {
-            notification.requestReplyId = requestReplyId;
+        if (cachedNotification) {
+            cachedNotification.requestReplyId = requestReplyId;
 
             // Bail early If @notificationParams represents an exact repeat
-            const title = notificationParams.title.unpack();
-            const body = notificationParams.body
-                ? notificationParams.body.unpack()
+            const title = notification.title;
+            const body = notification.body
+                ? notification.body
                 : null;
 
-            if (notification.title === title &&
-                notification.bannerBodyText === body) {
+            if (cachedNotification.title === title &&
+                cachedNotification.bannerBodyText === body) {
                 this._notificationPending = false;
                 return;
             }
 
-            notification.title = title;
-            notification.bannerBodyText = body;
+            cachedNotification.title = title;
+            cachedNotification.bannerBodyText = body;
+            cachedNotification.body = body;
 
         // Device Notification
         } else if (idMatch) {
-            notification = this._createNotification(notificationParams);
+            cachedNotification = notification;
 
-            notification.deviceId = deviceId;
-            notification.remoteId = remoteId;
-            notification.requestReplyId = requestReplyId;
+            cachedNotification.deviceId = deviceId;
+            cachedNotification.remoteId = remoteId;
+            cachedNotification.requestReplyId = requestReplyId;
 
-            notification.connect('destroy', (notification, reason) => {
+            cachedNotification.connect('destroy', (notification, reason) => {
                 this._closeGSConnectNotification(notification, reason);
                 delete this._notifications[localId];
             });
 
-            this._notifications[localId] = notification;
+            this._notifications[localId] = cachedNotification;
 
         // Service Notification
         } else {
-            notification = this._createNotification(notificationParams);
-            notification.connect('destroy', (notification, reason) => {
+            cachedNotification.connect('destroy', (notification, reason) => {
                 delete this._notifications[localId];
             });
-            this._notifications[localId] = notification;
+            this._notifications[localId] = cachedNotification;
         }
-
-        if (showBanner)
-            this.showNotification(notification);
-        else
-            this.pushNotification(notification);
-
-        this._notificationPending = false;
-    }
-
-    /*
-     * Override to raise the usual notification limit (3)
-     */
-    pushNotification(notification) {
-        if (this.notifications.includes(notification))
+        
```

Comment:
```suggestion

```

Reply:
Fixed :+1:

Reply:
Not really used to work with eslint :sweat_smile: